src/test/kotlin/apply/application/JudgeServiceTest.kt

```diff
@@ -0,0 +1,88 @@
+    val githubApi = mockk<GithubApi>()
+
+    val judgeService = JudgeService(
+        missionRepository, assignmentRepository, judgeHistoryRepository, githubApi
+    )
+
+    Given("과제 제출 시간이 아닐 때") {
+        every { missionRepository.getById(any()) } returns createMission(
+            startDateTime = LocalDateTime.now().minusMinutes(2),
+            endDateTime = LocalDateTime.now().minusMinutes(1)
+        )
+        every { githubApi.requestCommits(any()) } returns listOf(createCommitResponse())
+        every { assignmentRepository.getByUserIdAndMissionId(any(), any()) } returns createAssignment()
+        every { judgeHistoryRepository.findLastByUserIdAndMissionId(any(), any()) } returns null
+
+        When("예제 테스트를 실행하면") {
+            Then("예제 테스트가 실행되지 않는다.") {
+                shouldThrow<IllegalStateException> { judgeService.runExampleTest(1L, 1L) }
+            }
+        }
+    }
+
+    Given("과제 제출 마감 시간 이전의 가장 최신 커밋이 존재하지 않을 때") {
+        every { missionRepository.getById(any()) } returns createMission()
+        every { githubApi.requestCommits(any()) } returns listOf(
+            createCommitResponse(date = LocalDateTime.now().plusDays(8))
+        )
+        every { assignmentRepository.getByUserIdAndMissionId(any(), any()) } returns createAssignment()
+        every { judgeHistoryRepository.findLastByUserIdAndMissionId(any(), any()) } returns null
+
+        When("예제 테스트를 실행하면") {
+            Then("예제 테스트가 실행되지 않는다.") {
+                shouldThrow<IllegalArgumentException> { judgeService.runExampleTest(1L, 1L) }
+            }
+        }
+    }
+
+    Given("테스트 실행 이력이 존재할 때") {
+        val assignment = createAssignment()
+        every { missionRepository.getById(any()) } returns createMission()
+        every { githubApi.requestCommits(any()) } returns listOf(createCommitResponse())
+        every { assignmentRepository.getByUserIdAndMissionId(any(), any()) } returns assignment
+
+        When("제출한 commit id와 최신 이력의 commit id가 같은 예제 테스트를 실행하면") {
+            val existHistory = createJudgeHistory(commitHash = COMMIT_HASH)
+            every { judgeHistoryRepository.findLastByUserIdAndMissionId(any(), any()) } returns existHistory
+            Then("기존의 최신 테스트 실행 결과를 반환한다.") {
+                val executionResponse = judgeService.runExampleTest(1L, 1L)
+                executionResponse shouldBe JudgeHistoryResponse(existHistory, assignment)
+            }
+        }
+
+        When("제출한 commit id와 최신 이력의 commit id가 다른 예제 테스트를 실행하면") {
+            val existHistory = createJudgeHistory(commitHash = "old-commit")
+            every { judgeHistoryRepository.findLastByUserIdAndMissionId(any(), any()) } returns existHistory
+            Then("테스트를 실행한 뒤 새로운 테스트 실행 결과를 반환한다.") {
+                val executionResponse = judgeService.runExampleTest(1L, 1L)
+                executionResponse shouldNotBe JudgeHistoryResponse(existHistory, assignment)
+            }
+        }
```

Comment:
r: 마찬가지로 공백 컨벤션입니다.
When과 Then 사이에 코드가 존재한다면 공백을 두어 가독성을 높이는 방향으로 컨벤션이 잡혀있어요
```suggestion
        When("제출한 commit id와 최신 이력의 commit id가 같은 예제 테스트를 실행하면") {
            val existHistory = createJudgeHistory(commitHash = COMMIT_HASH)
            every { judgeHistoryRepository.findLastByUserIdAndMissionId(any(), any()) } returns existHistory
            
            Then("기존의 최신 테스트 실행 결과를 반환한다.") {
                val executionResponse = judgeService.runExampleTest(1L, 1L)
                executionResponse shouldBe JudgeHistoryResponse(existHistory, assignment)
            }
        }

        When("제출한 commit id와 최신 이력의 commit id가 다른 예제 테스트를 실행하면") {
            val existHistory = createJudgeHistory(commitHash = "old-commit")
            every { judgeHistoryRepository.findLastByUserIdAndMissionId(any(), any()) } returns existHistory
            
            Then("테스트를 실행한 뒤 새로운 테스트 실행 결과를 반환한다.") {
                val executionResponse = judgeService.runExampleTest(1L, 1L)
                executionResponse shouldNotBe JudgeHistoryResponse(existHistory, assignment)
            }
        }
```